Fix autoconf check for C99 and update build to run autoreconf.
https://sourceforge.net/p/mikmod/bugs/29/

diff --git a/configure.ac b/configure.ac
index b5cdbb5..7378cc0 100644
--- a/configure.ac
+++ b/configure.ac
@@ -608,10 +608,10 @@ then
 int main(void) {
 	void *libc, *printfptr;
 	libc = dlopen("libc.so", RTLD_LAZY | RTLD_GLOBAL);
-	if (!libc) exit(1);
+	if (!libc) return 1;
 	printfptr = dlsym(libc, "_printf");
 	dlclose(libc);
-	exit(!printfptr);
+	return !printfptr;
 }],
 			[libmikmod_cv_dynload_underscores=yes],
 			[libmikmod_cv_dynload_underscores=no],
